lib/matplotlib/scale.py

```diff
@@ -111,31 +93,35 @@ class LogTransformBase(Transform):
     def __init__(self, nonpos):
         Transform.__init__(self)
         if nonpos == 'mask':
-            self._handle_nonpos = _mask_non_positives
+            self._fill_value = np.nan
         else:
-            self._handle_nonpos = _clip_non_positives
-
-
-class Log10Transform(LogTransformBase):
-    base = 10.0
+            self._fill_value = 1e-300
 
     def transform_non_affine(self, a):
-        a = self._handle_nonpos(a * 10.0)
-        return np.log10(a)
-
-    def inverted(self):
-        return InvertedLog10Transform()
+        with np.errstate(invalid="ignore"):
+            a = np.where(a <= 0, self._fill_value, a)
+        return np.divide(np.log(a, out=a), np.log(self.base), out=a)
 
 
-class InvertedLog10Transform(Transform):
+class InvertedLogTransformBase(Transform):
     input_dims = 1
     output_dims = 1
     is_separable = True
     has_inverse = True
-    base = 10.0
 
     def transform_non_affine(self, a):
-        return ma.power(10.0, a) / 10.0
+        return ma.power(self.base, a)
+
+
+class Log10Transform(LogTransformBase):
+    base = 10.0
+
+    def inverted(self):
+        return InvertedLog10Transform()
```

Comment:
No, because InvertedLog10Transform is not defined yet at that point.  (InvertedLog10Transform could define `inverted` as a reference to the `Log10Transform` class but I decided to keep definitions symmetric.)